### ISSUE TITLE
Let `json_encode()`/`json_decode()` accept/return DS lists 

### ISSUE THREAD

**Issue by backYard321** [YoYoGames/GameMaker-Bugs#5443]
### Is your feature request related to a problem?

Currently, to create a JSON from a DS list, it has to be placed into a DS map first using something like `ds_map_add_list()`. This adds many additional steps which can theoretically be avoided now that maps/lists are handles with distinct refs which can be differentiated between, rather than just numbers. 

As well, `json_decode()` will return a map, which is also unnecessary in cases where a JSON is only being used to store a list.

### Describe the solution you'd like

Allow for `json_encode()` to accept DS lists, and by extension for `json_decode()` to return a DS list (depending on the contents of the JSON)

**Comment by thennothinghappened** [comment]
Currently if you `json_decode()` something like this:
```json
["hi"]
```
I'm pretty sure it gets taken in something like this:
```json
{ "default": ["hi"] }
```
And thus is guaranteed to always return a `ds_map`. In older versions of the engine when DS entries were just numbers, if it could return a list instead you would have no way of knowing whether it was one, or that you just happened to also have a `ds_list` entry at that index:
```gml
var some_other_list = ds_list_create();
// ...

var parsed_thing = json_decode(@'{ "hi": 2 }');

if (ds_exists(parsed_thing, ds_type_list)) {
    // this would run despite it being a map!
}
```

Today, we have refs instead (_or handles..._) and they can be differentiated by the engine, but `json_decode` is already a function that is used largely for legacy support, and changing its behaviour here would break that - all old code relying on the return value always being a map will break.

**Comment by backYard321** [comment]
> ["hi"]

Good point - I was under the wrong impression that it simply doesn't decode JSONs that aren't formatted as a map. So yeah, looks like returning a list is totally out of the question.
